File: file-processing/cpp/src/student.cpp
#include "student.hpp"

std::istream& operator >> (std::istream &input, Student &me) {
	//const bool is_terminal = (&input == &std::cin);
	input >> me.my_id;
	input.ignore(1, '\n');
	std::getline(input, me.registration_date);
	std::getline(input, me.first_name);
	std::getline(input, me.last_name);
	std::getline(input, me.document_type);
	std::getline(input, me.document);
	std::getline(input, me.phone);
	std::getline(input, me.email);
	input >> me.city_id;
	return input;
}

std::ostream& operator << (std::ostream &output, const Student &me) {
	const bool is_terminal = (&output == &std::cout);

	output << (is_terminal? "Id:                  " : "") <<
		me.my_id << '\n';

	output << (is_terminal? "Data de registro:    " : "") <<
		me.registration_date << '\n';
	
	output << (is_terminal? "Primeiro Nome:       " : "") <<
		me.first_name << '\n';
	
	output << (is_terminal? "Ultimo Nome:         " : "") <<
		me.last_name << '\n';
	
	output << (is_terminal? "Tipo do documento:   " : "") <<
		me.document_type << '\n';
	
	output << (is_terminal? "Numero do documento: " : "") <<
		me.document << '\n';
	
	output << (is_terminal? "Numero de telefone:  " : "") <<
		me.phone << '\n';
	
	output << (is_terminal? "Email:               " : "") <<
		me.email << '\n';
	
	output << (is_terminal? "Id da cidade:        " : "") <<
		me.city_id << '\n';
	return output;
}

File: file-processing/python/student.py
from dataclasses import dataclass
from io import TextIOWrapper

@dataclass()
class Student:
	my_id: int = 0
	registration_date: str = ''
	first_name: str = ''
	last_name: str = ''
	document_type: str = ''
	document: str = ''
	phone: str = ''
	email: str = ''
	city: str = ''

	def readFromTerminal():
		s = Student()
		s.first_name = input('Entre com o primeiro nome: ')
		s.last_name = input('Entre com o ultimo nome: ')
		s.document_type = input('Entre com o tipo de documento: ')
		s.document = input('Entre com os numeros do documento: ')
		s.phone = input('Entre com os numeros do telefone: ')
		s.email = input('Entre com o email: ')
		s.city = input('Entre com a cidade: ')
		return s

	def read(self, file: TextIOWrapper) -> bool:
		temp = file.readline()[:-1]
		if (not temp):
			return False

		self.my_id = int(temp)
		self.registration_date = file.readline()[:-1]
		self.first_name = file.readline()[:-1]
		self.last_name = file.readline()[:-1]
		self.document_type = file.readline()[:-1]
		self.document = file.readline()[:-1]
		self.phone = file.readline()[:-1]
		self.email = file.readline()[:-1]
		self.city = file.readline()[:-1]
		return True

	def write(self, file: TextIOWrapper) -> None:
		file.write(str(self.my_id) + '\n')
		file.write(self.registration_date + '\n')
		file.write(self.first_name + '\n')
		file.write(self.last_name + '\n')
		file.write(self.document_type + '\n')
		file.write(self.document + '\n')
		file.write(self.phone + '\n')
		file.write(self.email + '\n')
		file.write(self.city + '\n')

